src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -93,10 +93,11 @@ protected virtual string GetResourcePrefix(TypeInfo typeInfo, string baseNamespa
             {
                 throw new ArgumentNullException(nameof(baseNamespace));
             }
+            var assemblyName = new AssemblyName(typeInfo.Assembly.FullName).Name;
 
             return string.IsNullOrEmpty(resourcesRelativePath)
                 ? typeInfo.FullName
-                : baseNamespace + "." + resourcesRelativePath + TrimPrefix(typeInfo.FullName, baseNamespace + ".");
+                : baseNamespace + "." + resourcesRelativePath + TrimPrefix(typeInfo.FullName, assemblyName + ".");
```

Comment:
Why does this method now ignore `baseNamespace` (except for the `null` or empty check)?

Reply:
Spoke with @dougbu in person and we now agree this is correct.